mediapipe/tasks/ios/test/vision/gesture_recognizer/MPPGestureRecognizerTests.m

```diff
@@ -0,0 +1,280 @@
+  for (int i = 0; i < expectedTopHandGestures.count; i++) {
+    MPPCategory *gesture = topHandGestures[i];
+    XCTAssertNotNil(gesture);
+    AssertEqualGestures(gesture, expectedTopHandGestures[i], 0, i);
+  }
+}
+
+- (void)assertGestureRecognizerResult:(MPPGestureRecognizerResult *)gestureRecognizerResult
+    isApproximatelyEqualToExpectedResult:
+        (MPPGestureRecognizerResult *)expectedGestureRecognizerResult {
+  [self assertMultiHandLandmarks:gestureRecognizerResult.landmarks
+      isApproximatelyEqualToExpectedMultiHandLandmarks:expectedGestureRecognizerResult.landmarks];
+  [self assertMultiHandWorldLandmarks:gestureRecognizerResult.worldLandmarks
+      isApproximatelyEqualToExpectedMultiHandWorldLandmarks:expectedGestureRecognizerResult
+                                                                .worldLandmarks];
+  [self assertMultiHandGestures:gestureRecognizerResult.gestures
+      isApproximatelyEqualToExpectedMultiHandGestures:expectedGestureRecognizerResult.gestures];
+}
+
+#pragma mark File
+
++ (NSString *)filePathWithFileInfo:(NSDictionary *)fileInfo {
+  NSString *filePath = [MPPGestureRecognizerTests filePathWithName:fileInfo[@"name"]
+                                                         extension:fileInfo[@"type"]];
+  return filePath;
+}
+
++ (NSString *)filePathWithName:(NSString *)fileName extension:(NSString *)extension {
+  NSString *filePath = [[NSBundle bundleForClass:self.class] pathForResource:fileName
+                                                                      ofType:extension];
+  return filePath;
+}
+
+#pragma mark Gesture Recognizer Initializers
+
+- (MPPGestureRecognizerOptions *)gestureRecognizerOptionsWithModelFileInfo:
+    (NSDictionary *)modelFileInfo {
+  NSString *modelPath = [MPPGestureRecognizerTests filePathWithFileInfo:modelFileInfo];
+  MPPGestureRecognizerOptions *gestureRecognizerOptions =
+      [[MPPGestureRecognizerOptions alloc] init];
+  gestureRecognizerOptions.baseOptions.modelAssetPath = modelPath;
+
+  return gestureRecognizerOptions;
+}
+
+- (MPPGestureRecognizer *)createGestureRecognizerWithOptionsSucceeds:
+    (MPPGestureRecognizerOptions *)gestureRecognizerOptions {
+  MPPGestureRecognizer *gestureRecognizer =
+      [[MPPGestureRecognizer alloc] initWithOptions:gestureRecognizerOptions error:nil];
+  XCTAssertNotNil(gestureRecognizer);
+
+  return gestureRecognizer;
+}
+
+- (void)assertCreateGestureRecognizerWithOptions:
+            (MPPGestureRecognizerOptions *)gestureRecognizerOptions
+                          failsWithExpectedError:(NSError *)expectedError {
+  NSError *error = nil;
+  MPPGestureRecognizer *gestureRecognizer =
+      [[MPPGestureRecognizer alloc] initWithOptions:gestureRecognizerOptions error:&error];
+
+  XCTAssertNil(gestureRecognizer);
+  AssertEqualErrors(error, expectedError);
+}
+
+#pragma mark Assert Gesture Recognizer Results
+
+- (MPPImage *)imageWithFileInfo:(NSDictionary *)fileInfo {
```

Comment:
FWIW, we should eventually move all the shared code to a helpers file.

Reply:
Will do while writing tests for the image segmenter and slowly migrate all tests.